Assets/MRTK/Tools/MSBuild/Scripts/UnityProjectInfo.cs

```diff
@@ -238,14 +248,31 @@ private CSProjectInfo GetProjectInfo(Dictionary<string, CSProjectInfo> projectsM
 
             // Manually add special plugin dependencies to the projects
 #if UNITY_2019_3_OR_NEWER
+#if UNITY_2020_2_OR_NEWER
+            if (toReturn.Name.StartsWith("Microsoft.MixedReality.Toolkit") || toReturn.Name.StartsWith("Unity.TextMeshPro"))
+#else
             if (toReturn.Name.StartsWith("Microsoft.MixedReality.Toolkit"))
+#endif
             {
-                foreach (var plugin in SpecialPluginNameMappingUnity2019.Values)
+                string[] plugins = SpecialPluginNameMappingUnity2019.Values.OrderByDescending(p => p).ToArray();
+                foreach (var plugin in plugins)
                 {
                     if (projectsMap.TryGetValue(plugin, out CSProjectInfo projectInfo))
                     {
                         toReturn.AddDependency(projectInfo);
                     }
+#if UNITY_2020_2_OR_NEWER
+                    else
+                    {
+                        CSProjectInfo newProjInfo = new CSProjectInfo(this, asmDefInfoMap[plugin], projectOutputPath);
+                        if (plugin == plugins[1])
+                        {
+                            newProjInfo.AddDependency(projectsMap[plugins[0]]);
+                        }
+                        projectsMap.Add(plugin, newProjInfo);
+                        toReturn.AddDependency(newProjInfo);
+                    }
+#endif
```

Comment:
What is this block doing?

Reply:
It seems odd that we'd get to this point and not have all the projects we need in the map already

Reply:
This block creates UGUI csprojs and add them as dependency to the csproj being processed. I agree that it seems odd but with recent major updates of Unity (i.e. 2018 -> 2019 and 2019 -> 2020) UGUI seems to require a little extra handling.